File: solutions/fortran/isogram/1/isogram.f90
module isogram
  implicit none

contains

  function isIsogram(phrase) result(no_repeats)
    character(len=*), intent(in) :: phrase
    character(len=len(phrase)) :: sentence
    integer :: i,j
    logical :: no_repeats
    no_repeats=.true.
    sentence=phrase
    if (sentence=="") then
      no_repeats=.true.
    else
      do i=1,len(sentence)
        do j=i+1,len(sentence)
          if (sentence(i:i)/="-" .and. sentence(i:i)/=" ") then
            if (ichar(sentence(i:i))>=65 .and. ichar(sentence(i:i))<=90) then
              sentence(i:i)=char(ichar(sentence(i:i))+32)
            else if (ichar(sentence(j:j))>=65 .and. ichar(sentence(j:j))<=90) then
              sentence(j:j)=char(ichar(sentence(j:j))+32)
            end if
              no_repeats=no_repeats .and. (sentence(i:i)/=sentence(j:j))
          end if
        end do
      end do
    end if
  end function isIsogram

end module isogram

File: solutions/c/isogram/1/isogram.c
#include "isogram.h"
#include <string.h>
#include <ctype.h>

bool is_isogram(const char phrase[]){
    const char *pPtr = phrase;
    int i,j,lon;
    //check for null character
    bool isIso=true;
    if (pPtr==NULL){
        isIso=false;
    }
    //check for zero size string
    else if (strlen(phrase)==0){
        isIso=true;
    }
    else {
        lon=strlen(phrase);
        for (i=0;i<lon;i++){
            for (j=0;j<lon;j++){
                //for characters in different position that are letters and converted to lowercase
                //if the characters are equal, the word is not an isogram
                if(i!=j && tolower(phrase[i])==tolower(phrase[j]) && isalpha(phrase[i])){
                    isIso=false;
                    i=lon+1;
                    j=lon+1;
                }
            }
        }
    }
    return isIso;
}
